File: JokiBackend/src/main/java/unical/enterprise/jokibackend/Data/Entities/Review.java
package unical.enterprise.jokibackend.Data.Entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@Entity(name = "reviews")
@AllArgsConstructor
@NoArgsConstructor
public class Review {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.UUID)
    @Column
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "game_id", referencedColumnName = "id")
    private Game game;

    @Column
    @Size(min = 3, max = 300, message = "La recensione deve essere lunga almeno 3 caratteri")
    private String review;

    @Column
    private Boolean suggested;
}

File: JokiFrontend/src/app/model/review.ts
import {User} from "./user";
import {Game} from "./game";

export class Review {
  private gameId: string;
  private review: string;
  private suggested: boolean;

  constructor(gameId: string, review: string, suggested: boolean) {
    this.gameId = gameId;
    this.review = review;
    this.suggested = suggested;
  }

  public getGameId(): string {
    return this.gameId;
  }

  public setGameId(gameId: string): void {
    this.gameId = gameId;
  }

  public getReview(): string {
    return this.review;
  }

  public setReview(review: string): void {
    this.review = review;
  }

  public getSuggested(): boolean {
    return this.suggested;
  }

  public setSuggested(suggested: boolean): void {
    this.suggested = suggested;
  }
}
